rust/sealevel/libraries/ecdsa-signature/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+use hyperlane_core::H160;
+use solana_program::{
+    keccak,
+    secp256k1_recover::{secp256k1_recover, Secp256k1RecoverError},
+};
+
+/// Errors relating to an EcdsaSignature
+#[derive(Copy, Clone, Debug, Eq, thiserror::Error, PartialEq)]
+pub enum EcdsaSignatureError {
+    #[error("Invalid signature length")]
+    InvalidLength,
+    #[error("Invalid signature recovery ID")]
+    InvalidRecoveryId,
+}
+
+/// An ECDSA signature with a recovery ID.
+/// Signature recovery functions expect a 64 byte serialized r & s value and a 1 byte recovery ID
+/// that is either 0 or 1.
+/// This type is used to deserialize a 65 byte signature & allow easy recovery of the Ethereum
+/// address signer.
+#[derive(Debug, Eq, PartialEq)]
+pub struct EcdsaSignature {
```

Comment:
is there no readily available ecdsa signature library in solana?

Reply:
Not really - they have https://docs.rs/solana-program/latest/solana_program/secp256k1_recover/fn.secp256k1_recover.html which we use, and then this struct is a nicer interface into using that